### PR TITLE
feat(app-builder): initial WIP

### PR DESCRIPTION
**Pull request by kapetr**

### CODE REVIEW

File: src/modules/chat/message/markdown-plugins/remarkPythonAppCode.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Root } from 'hast';
+import { Code } from 'mdast';
+import { visit } from 'unist-util-visit';
+
+export const ATTACHMENT_FILE_PREFIX = 'urn:bee:file:';
```
Comment (PetrBulanek) [inline src/modules/chat/message/markdown-plugins/remarkPythonAppCode.ts:21]: I guess, this should not be here. :-)

File: src/modules/apps/manage/EditAppModal.tsx [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+  });
+
+  useEffect(() => {
+    if (!isEmpty(dirtyFields))
+      setConfirmOnRequestClose(
+        'Your app has unsaved changes, do you really want to leave?',
+      );
+    else clearConfirmOnRequestClose();
+    return () => clearConfirmOnRequestClose();
+  }, [clearConfirmOnRequestClose, dirtyFields, setConfirmOnRequestClose]);
```
Comment (PetrBulanek) [inline src/modules/apps/manage/EditAppModal.tsx:97]: Almost identical code is inside UserToolModal, so we could maybe abstract it into separate hook. But it's not really necessary right now.

**Review by PetrBulanek — APPROVED** [review]

File: .gitignore [hunk 3]
```diff
@@ -40,3 +40,6 @@ next-env.d.ts
 
 # IDEA
 .idea
+
+# stlite app
+public/stlite/dist
```
[the remark this answers is not in the document]
Comment (kapetr) [reply .gitignore]: Already removed, just forgot to push it.

File: .prettierignore [hunk 4]
```diff
@@ -2,7 +2,6 @@ node_modules/
 .vscode/
 .next/
 .yarn/
-public/
```
[the remark this answers is not in the document]
Comment (kapetr) [reply .prettierignore]: Also already reverted.

File: src/layout/providers/ModalControlProvider.tsx [hunk 5]
```diff
@@ -47,6 +47,7 @@ export function ModalControlProvider({
   const [confirmMessage, setConfirmMessage] = useState<string | null>(null);
 
   const setConfirmOnRequestClose = (confirmMessage?: string) => {
+    console.trace();
```
[the remark this answers is not in the document]
Comment (kapetr) [reply src/layout/providers/ModalControlProvider.tsx:50]: thanks!

File: src/modules/apps/UserContentFrame.tsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+  }, [getCode, state, triggerCodeUpdate]);
+
+  useEffect(() => {
+    const handleStliteMessage = (e: MessageEvent<StliteMessage>) => {
+      if (e.data.type === STLITE_MESSAGE_TYPE_STATE_CHANGED) {
+        if (e.data.scriptRunState === 'running') setState('ready');
+      }
+    };
+
+    window.removeEventListener('message', handleStliteMessage);
```
[the remark this answers is not in the document]
Comment (kapetr) [reply src/modules/apps/UserContentFrame.tsx:58]: Not anymore, removing, thanks.

File: src/modules/apps/UserContentFrame.tsx [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+  useEffect(() => {
+    if (state === 'ready') {
+      const code = getCode();
+      if (code) triggerCodeUpdate(getCode());
+    }
+  }, [getCode, state, triggerCodeUpdate]);
+
+  useEffect(() => {
+    const handleStliteMessage = (e: MessageEvent<StliteMessage>) => {
+      if (e.data.type === STLITE_MESSAGE_TYPE_STATE_CHANGED) {
```
[the remark this answers is not in the document]
Comment (kapetr) [reply src/modules/apps/UserContentFrame.tsx:53]: Let's revise this in subsequent PR, adding TODO comment.

**Pull request merged**